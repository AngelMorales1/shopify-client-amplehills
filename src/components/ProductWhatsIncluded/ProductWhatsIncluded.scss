@import 'settings';

.ProductWhatsIncluded {
  padding: spacing(10) spacing(4);
  flex-direction: column;
  align-items: center;

  @media ($medium-up) {
    padding: spacing(20) spacing(4) spacing(10);
    flex-direction: row;
    justify-content: flex-end;
  }

  &__block-title {
    @media ($medium-up) {
      position: absolute;
      left: 0;
      transform: translate(0, spacing(-30));
    }
  }

  &__illustration {
    @media ($medium-up) {
      position: absolute;
      left: 0;
      transform: translate(0, spacing(30));
    }
  }

  &__flavor-container {
    @media ($medium-up) {
      padding: spacing(2) 0;
      max-width: spacing(150);
    }
  }

  &--light-pink {
    @extend .bg-baby-pink;
    @extend .text-madison-blue;
  }
}
