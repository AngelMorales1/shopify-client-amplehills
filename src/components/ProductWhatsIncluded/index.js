import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import get from 'utils/get';
import productModel from 'models/productModel';
import imageModel from 'models/imageModel';
import contentfulImgUtil from 'utils/contentfulImgUtil';

import styles from './ProductWhatsIncluded.scss';
import { Image } from 'components/base';

const ProductWhatsIncluded = ({ block, z, products }) => {
  const fields = get(block, 'fields', {});
  const colorClass = `ProductWhatsIncluded--${get(
    fields,
    'backgroundColor',
    'light-pink'
  )}`;

  const includedItems = get(fields, 'products', []);

  return (
    <div
      className={cx(
        styles['ProductWhatsIncluded'],
        styles[colorClass],
        'flex justify-between drip relative'
      )}
      style={{ zIndex: z }}
    >
      <h2
        className={cx(
          styles['ProductWhatsIncluded__block-title'],
          'block-headline my3 nowrap col-12 md-col-6 center'
        )}
      >
        What&rsquo;s included
      </h2>
      <div className="flex flex-column col-12 md-col-6">
        {includedItems.map(includedItem => {
          const handle = get(includedItem, 'fields.productHandle', '');
          const product = get(products, handle, {});

          return (
            <div
              key={handle}
              className={cx(
                styles['ProductWhatsIncluded__flavor-container'],
                'flex items-start my2 col-12'
              )}
            >
              <div className="mr3 col-2">
                <Image
                  alt={`${product.title} image`}
                  src={contentfulImgUtil(product.pintImage, '500', 'png')}
                />
              </div>
              <div className="col-10">
                <h3 className="description-title bold mb2">{`1x ${
                  product.title
                }`}</h3>
                <p className="block-subheadline">{product.flavorDescription}</p>
              </div>
            </div>
          );
        })}
      </div>
      {fields.illustration ? (
        <div
          className={cx(
            styles['ProductWhatsIncluded__illustration'],
            'col-12 md-col-6 center'
          )}
        >
          <Image
            className="col-5 md-col-4 mt3"
            alt="what&rsquo;s included image"
            src={contentfulImgUtil(
              get(fields, 'illustration.fields.file.url', ''),
              '600',
              'png'
            )}
          />
        </div>
      ) : null}
    </div>
  );
};

export default ProductWhatsIncluded;

ProductWhatsIncluded.propTypes = {
  block: PropTypes.shape({
    fields: PropTypes.shape({
      backgroundColor: PropTypes.string,
      products: PropTypes.arrayOf(
        PropTypes.shape({
          fields: PropTypes.shape({
            productHandle: PropTypes.string
          })
        })
      ),
      illustration: imageModel.propTypes
    })
  }),
  z: PropTypes.number,
  products: PropTypes.objectOf(productModel.propTypes)
};

ProductWhatsIncluded.defaultProps = {
  block: {
    fields: {
      backgroundColor: 'light-pink',
      products: [
        {
          fields: {
            productHandle: ''
          }
        }
      ],
      illustration: null
    }
  },
  z: 1,
  products: {}
};
