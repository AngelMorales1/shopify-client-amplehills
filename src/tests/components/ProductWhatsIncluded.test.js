import React from 'react';
import { shallow } from 'enzyme';

import ProductWhatsIncluded from 'components/ProductWhatsIncluded';

it('renders without data', () => {
  const component = shallow(<ProductWhatsIncluded />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <ProductWhatsIncluded
      block={{
        backgroundColor: 'foo',
        products: [
          {
            fields: {
              productHandle: 'foo'
            }
          }
        ],
        title: 'foo'
      }}
      product={{
        available: true,
        blocks: [],
        cartDetails: 'foo',
        description: 'foo',
        flavorDescription: 'foo',
        gridImage: 'foo',
        handle: 'foo',
        id: 'foo',
        pintImage: 'foo',
        price: 0,
        subItems: [],
        subItemsAvailable: true,
        title: 'foo',
        variants: [{}]
      }}
      z={0}
    />
  );

  expect(component).toMatchSnapshot();
});
