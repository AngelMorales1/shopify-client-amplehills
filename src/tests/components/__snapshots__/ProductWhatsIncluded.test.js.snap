// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductWhatsIncluded
    block={
        Object {
            "backgroundColor": "foo",
            "products": Array [
              Object {
                "fields": Object {
                  "productHandle": "foo",
                },
              },
            ],
            "title": "foo",
          }
    }
    product={
        Object {
            "available": true,
            "blocks": Array [],
            "cartDetails": "foo",
            "description": "foo",
            "flavorDescription": "foo",
            "gridImage": "foo",
            "handle": "foo",
            "id": "foo",
            "pintImage": "foo",
            "price": 0,
            "subItems": Array [],
            "subItemsAvailable": true,
            "title": "foo",
            "variants": Array [
              Object {},
            ],
          }
    }
    products={Object {}}
    setRef={[Function]}
    whatsIncluded={
        Object {
            "whatsIncludedDrip": false,
            "whatsIncludedIllustration": "",
            "whatsIncludedProducts": Array [],
          }
    }
    wholesaleProducts={Object {}}
    z={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-12 md-col-6"
>
          <h2
                    className="block-headline my3 nowrap center flex flex-column items-center justify-center"
          >
                    Available Flavors
          </h2>
</div>,
        <div
          className="flex flex-column col-12 md-col-6"
>
          
</div>,
        null,
      ],
      "className": "flex justify-between relative",
      "style": Object {
        "zIndex": 0,
      },
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="block-headline my3 nowrap center flex flex-column items-center justify-center"
>
              Available Flavors
</h2>,
            null,
          ],
          "className": "col-12 md-col-6",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Available Flavors",
              "className": "block-headline my3 nowrap center flex flex-column items-center justify-center",
            },
            "ref": null,
            "rendered": "Available Flavors",
            "type": "h2",
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "flex flex-column col-12 md-col-6",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-12 md-col-6"
>
            <h2
                        className="block-headline my3 nowrap center flex flex-column items-center justify-center"
            >
                        Available Flavors
            </h2>
</div>,
          <div
            className="flex flex-column col-12 md-col-6"
>
            
</div>,
          null,
        ],
        "className": "flex justify-between relative",
        "style": Object {
          "zIndex": 0,
        },
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="block-headline my3 nowrap center flex flex-column items-center justify-center"
>
                Available Flavors
</h2>,
              null,
            ],
            "className": "col-12 md-col-6",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Available Flavors",
                "className": "block-headline my3 nowrap center flex flex-column items-center justify-center",
              },
              "ref": null,
              "rendered": "Available Flavors",
              "type": "h2",
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "flex flex-column col-12 md-col-6",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductWhatsIncluded
    block={
        Object {
            "fields": Object {
              "backgroundColor": "light-pink",
              "illustration": null,
              "products": Array [
                Object {
                  "fields": Object {
                    "productHandle": "",
                  },
                },
              ],
            },
          }
    }
    products={Object {}}
    setRef={[Function]}
    whatsIncluded={
        Object {
            "whatsIncludedDrip": false,
            "whatsIncludedIllustration": "",
            "whatsIncludedProducts": Array [],
          }
    }
    wholesaleProducts={Object {}}
    z={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-12 md-col-6"
>
          <h2
                    className="block-headline my3 nowrap center flex flex-column items-center justify-center"
          >
                    Available Flavors
          </h2>
</div>,
        <div
          className="flex flex-column col-12 md-col-6"
>
          
</div>,
        null,
      ],
      "className": "flex justify-between relative",
      "style": Object {
        "zIndex": 1,
      },
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="block-headline my3 nowrap center flex flex-column items-center justify-center"
>
              Available Flavors
</h2>,
            null,
          ],
          "className": "col-12 md-col-6",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Available Flavors",
              "className": "block-headline my3 nowrap center flex flex-column items-center justify-center",
            },
            "ref": null,
            "rendered": "Available Flavors",
            "type": "h2",
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "flex flex-column col-12 md-col-6",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-12 md-col-6"
>
            <h2
                        className="block-headline my3 nowrap center flex flex-column items-center justify-center"
            >
                        Available Flavors
            </h2>
</div>,
          <div
            className="flex flex-column col-12 md-col-6"
>
            
</div>,
          null,
        ],
        "className": "flex justify-between relative",
        "style": Object {
          "zIndex": 1,
        },
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="block-headline my3 nowrap center flex flex-column items-center justify-center"
>
                Available Flavors
</h2>,
              null,
            ],
            "className": "col-12 md-col-6",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Available Flavors",
                "className": "block-headline my3 nowrap center flex flex-column items-center justify-center",
              },
              "ref": null,
              "rendered": "Available Flavors",
              "type": "h2",
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "flex flex-column col-12 md-col-6",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
